src/context/roiReducer.tsx

```diff
@@ -158,6 +159,15 @@ export function roiReducer(
         return;
       }
 
+      case 'UNSELECT_ROI': {
+        const id = action.payload;
+        const index = draft.rois.findIndex((roi) => roi.id === id);
+        if (index === -1) return;
+
+        draft.selectedRoi = undefined;
+        return;
+      }
```

Comment:
i'm not really sure if i really need to use "id" since, only one roi can be selected

Reply:
I think what you want to do here is
```js
if(draft.selectedRoi === action.payload) {
  draft.selectedRoi = undefined;
}
```